resources/google_drive_upload/drive.py

```diff
@@ -0,0 +1,114 @@
+import httplib2
+import os
+from dotmap import DotMap
+
+from apiclient import discovery, errors
+from apiclient.http import MediaFileUpload
+from oauth2client import client
+from oauth2client import tools
+from oauth2client.file import Storage
+
+# If modifying these scopes, delete your previously saved credentials
+# at ~/.credentials/ml_template_gdrive.json
+SCOPES = "https://www.googleapis.com/auth/drive.file"
+APPLICATION_NAME = "Drive API Python Quickstart"
+DEBUG = False
+
+
+def get_credentials():
+    """Gets valid user credentials from storage.
+
+    If nothing has been stored, or if the stored credentials are invalid,
+    the OAuth2 flow is completed to obtain the new credentials.
+
+    Returns:
+        Credentials, the obtained credential.
+    """
+    home_dir = os.path.expanduser("~")
+    credential_dir = os.path.join(home_dir, ".credentials")
+    if not os.path.exists(credential_dir):
+        os.makedirs(credential_dir)
+    credential_path = os.path.join(credential_dir, "ml_template_gdrive.json")
+
+    store = Storage(credential_path)
+    credentials = store.get()
+    if not credentials or credentials.invalid:
+        flow = client.flow_from_clientsecrets(CLIENT_SECRET_FILE, SCOPES)
+        flow.user_agent = APPLICATION_NAME
+        flags = DotMap()
+        flags.noauth_local_webserver = True
+        flags.logging_level = "ERROR"
+        credentials = tools.run_flow(flow, store, flags)
+        if DEBUG:
+            print("Storing credentials to " + credential_path)
+    return credentials
+
+
+def delete_item(service, item):
+    assert item["name"].endswith(".ipynb")
+    print("    Deleting: {0} ({1})".format(item["name"], item["id"]))
+    try:
+        service.files().delete(fileId=item["id"]).execute()
+    except errors.HttpError as error:
+        print("    An error occurred: {}".format(error))
+
+
+def delete_all_in_folder(service, folder_id):
+    PAGE_SIZE = 100
+    query = "'{}' in parents".format(folder_id)
+    if DEBUG:
+        print("Query:", query)
+    results = service.files().list(
+        spaces="drive", q=query, pageSize=PAGE_SIZE, fields="nextPageToken, files(id, name)").execute()
+    items = results.get("files", [])
+    if not items:
+        print("  No files found.")
+    else:
+        print("  Files:")
```

Comment:
This doesn't print the file name, right??

Reply:
It does, see line 49

Reply:
Example:
```
Deleting old notebook versions...
  Files:
    Deleting: mnist_gan_graph.ipynb (1KTymIekfqkIb3F1zw0sr1WOAcVckznr-)
    Deleting: mnist_cnn_keras.ipynb (1b6SX_iDxB-r-YBGx8FU0sstH3XRcB9eW)
    Deleting: mnist_cnn_eager.ipynb (1_HLzzi3xDi-r9bgnaNTYuD3xgmSBHa6p)
    Deleting: test.ipynb (1J0ULB1FtT3kqKC_Dnpisrx_g93mIzhlf)

Uploading new notebook versions...
  Uploading notebooks/test.ipynb
    File ID: 19xXIXsloHnMIk0-4V8GyBYWKtn1u30_l
  Uploading notebooks/mnist_cnn_eager.ipynb
    File ID: 1X25o0--XO2LyVHF-Jrolf5NXWZth1YMp
  Uploading notebooks/mnist_cnn_keras.ipynb
    File ID: 1kIZpZgXKgM76ihFzPtyHQjQO_TV3bECL
  Uploading notebooks/mnist_gan_graph.ipynb
    File ID: 1W0K0BY4W_-pJbm4eQhKzGYIwHWT14xdr
```